### PR TITLE
PixelPaint: StatusBar and ColorPicker update when moving the cursor

### PR DESCRIPTION
**Pull request by skmagiik**
MainWidget and ImageEditor now expose StatusBar append messages so tools can append messages after the ImageEditor cursor location coordinates.

ColorPIcker also has onmove support to live render the colors that you are selecting.

### CODE REVIEW

File: Userland/Applications/PixelPaint/ImageEditor.cpp [hunk 1]
```diff
@@ -318,13 +318,15 @@ void ImageEditor::set_editor_color_to_color_at_mouse_position(GUI::MouseEvent co
...
+
     // We picked a transparent pixel, do nothing.
     if (!color.alpha())
         return;
 
-    if (event.button() == GUI::MouseButton::Primary)
+    if (event.buttons() & GUI::MouseButton::Primary)
         set_primary_color(color);
-    else if (event.button() == GUI::MouseButton::Secondary)
+    else if (event.buttons() & GUI::MouseButton::Secondary)
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/PixelPaint/ImageEditor.cpp:329]: I don't see the point of this change. We have an else-if, so the cases are still exclusive. If you want to allow both to happen at the same time, remove the else, but this change is unrelated to everything else in the commit. You can add it as another commit on this PR to keep the commit history clean.

File: Userland/Applications/PixelPaint/Tools/PickerTool.cpp [hunk 2]
```diff
@@ -22,6 +22,21 @@ void PickerTool::on_mousedown(Layer* layer, MouseEvent& event)
...
+
+void PickerTool::on_mousemove(Layer* layer, MouseEvent& event)
+{
+    if (!layer)
+        return;
+    auto layer_event = event.layer_event();
+    if (!(layer_event.buttons() & GUI::MouseButton::Primary || layer_event.buttons() & GUI::MouseButton::Secondary))
+        return;
+    m_editor->set_editor_color_to_color_at_mouse_position(layer_event, m_sample_all_layers);
+}
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/PixelPaint/Tools/PickerTool.cpp:38]: This change, again, is unrelated. You're allowing color picker to continously pick a new color while the mouse is being pressed, but that is not related to the status bar changes.

File: Userland/Applications/PixelPaint/ImageEditor.cpp (hunk 1 above)
Comment (skmagiik) [reply Userland/Applications/PixelPaint/ImageEditor.cpp:329]: The reason for this change is on a move event it was not matching the equals conditions and was failing to update while moving the mouse.

File: Userland/Applications/PixelPaint/Tools/PickerTool.cpp (hunk 2 above)
Comment (skmagiik) [reply Userland/Applications/PixelPaint/Tools/PickerTool.cpp:41]: Updated to two separate commits.

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
Please mark comments as resolved once you have addressed them so I see you have dealt with them.

File: Userland/Applications/PixelPaint/MainWidget.cpp [hunk 3]
```diff
@@ -1175,4 +1180,18 @@ void MainWidget::drop_event(GUI::DropEvent& event)
...
+{
+    if (!appended_text.is_empty()) {
+        StringBuilder builder = StringBuilder();
+        builder.append(m_last_image_editor_mouse_position.to_deprecated_string());
+        builder.append(" "sv);
+        builder.append(appended_text);
+        m_statusbar->set_override_text(builder.to_deprecated_string());
+    } else {
+        m_statusbar->set_override_text(m_last_image_editor_mouse_position.to_deprecated_string());
+    }
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/PixelPaint/MainWidget.cpp:1194]: What I mean is that you should create a StringBuilder outside the if-else, append the mouse position to it, then only append the appended_text if it's nonempty, then set the override_text to the string builder result in either case. Therefore, the if is just one small subcase of the general case which shares most of the code.

File: Userland/Applications/PixelPaint/MainWidget.cpp [hunk 4]
```diff
@@ -1080,14 +1080,19 @@ ImageEditor& MainWidget::create_new_editor(NonnullRefPtr<Image> image)
...
         if (image_rectangle.contains(mouse_position)) {
-            m_statusbar->set_override_text(mouse_position.to_deprecated_string());
+            update_status_bar(current_image_editor()->appended_status_info());
             m_histogram_widget->set_color_at_mouseposition(current_image_editor()->image().color_at(mouse_position));
             m_vectorscope_widget->set_color_at_mouseposition(current_image_editor()->image().color_at(mouse_position));
         } else {
             m_statusbar->set_override_text({});
             m_histogram_widget->set_color_at_mouseposition(Color::Transparent);
             m_vectorscope_widget->set_color_at_mouseposition(Color::Transparent);
         }
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/PixelPaint/MainWidget.cpp:1090]: Same here, again :^)

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
I hope it's clear what you should do about the builder code, other than that this looks good

File: Userland/Applications/PixelPaint/MainWidget.cpp (hunk 3 above)
Comment (skmagiik) [reply Userland/Applications/PixelPaint/MainWidget.cpp]: Corrected, I understand what you mean now

File: Userland/Applications/PixelPaint/MainWidget.cpp (hunk 4 above)
Comment (skmagiik) [reply Userland/Applications/PixelPaint/MainWidget.cpp:1090]: Unclear to me what you mean here now.
The update_status_bar will check if the appended text is empty or not and then update the text.
Is more optimal to check if appended status info is empty here and avoid that function call?

Comment (kleinesfilmroellchen) [reply Userland/Applications/PixelPaint/MainWidget.cpp:1127]: Looking at it more closely, this situation is a little different so it's fine here

**Review by kleinesfilmroellchen — APPROVED** [review]
